plugins/inputs/dmcache/dmcache_linux.go

```diff
@@ -0,0 +1,131 @@
+// +build linux
+
+package dmcache
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+)
+
+const metricName = "dmcache"
+
+var fieldNames = [...]string{
+	"metadata_used",
+	"metadata_free",
+	"cache_used",
+	"cache_free",
+	"read_hits",
+	"read_misses",
+	"write_hits",
+	"write_misses",
+	"demotions",
+	"promotions",
+	"dirty",
+}
+
+func (c *DMCache) Gather(acc telegraf.Accumulator) error {
+	outputLines, err := c.rawStatus()
+	if err != nil {
+		return err
+	}
+
+	var total map[string]interface{}
+	if !c.PerDevice {
+		total = make(map[string]interface{})
+	}
+
+	for _, s := range outputLines {
+		fields := make(map[string]interface{})
+		data, err := parseDMSetupStatus(s)
+		if err != nil {
+			return err
+		}
+
+		for _, f := range fieldNames {
+			fields[f] = calculateSize(data, f)
+		}
+
+		if c.PerDevice {
```

Comment:
this if-statement doesn't seem correct to me.

If the user sets per_device = true, then it should report the per-device stats AND total stats.

Is there a reason users would not want total stats? If so it should be a separate config option.